### PR TITLE
meson CI additions

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

**Comment by neheb** [comment]
```
../src/tiffimage_int.cpp(1957): fatal error C1001: An internal error has occurred in the compiler.
```

...interesting

File: src/asfvideo.cpp [hunk 1]
```diff
@@ -83,13 +83,13 @@ std::string AsfVideo::GUIDTag::to_string() {
 bool AsfVideo::GUIDTag::operator<(const GUIDTag& other) const {
   if (data1_ < other.data1_)
     return true;
-  else if (data1_ == other.data1_) {
+  if (data1_ == other.data1_) {
     if (data2_ < other.data2_)
       return true;
-    else if (data2_ == other.data2_) {
+    if (data2_ == other.data2_) {
       if (data3_ < other.data3_)
```
Comment (piponazo) [inline src/asfvideo.cpp:90]: Would you mind to take this opportunity to add the brackets were they are missing? It looks super confusing to have some ifs/elses with them and other without.

File: src/image.cpp [hunk 2]
```diff
@@ -733,19 +733,15 @@ AccessMode ImageFactory::checkMode(ImageType type, MetadataId metadataId) {
...
-      return r->xmpSupport_;
-    case mdComment:
-      return r->commentSupport_;
-    default:
-      return amNone;
-  }
-  return {};
+  if (metadataId == mdExif)
+    return r->exifSupport_;
+  if (metadataId == mdIptc)
```
Comment (piponazo) [inline src/image.cpp:738]: [question] I see why you are making this change (I have been there and done this 😛 ) . However, would it not make more sense to have `else if` statements after the first if  ?

**Review by piponazo — APPROVED** [review]

File: src/image.cpp (hunk 2 above)
Comment (neheb) [reply src/image.cpp:738]: Not really. else after return is redundant.

**Comment by piponazo** [comment]
I see that many of the changes here are also present in #2508. Is this intended or was it an accident? Which PR do you plan to merge first?

**Comment by neheb** [comment]
I'll merge this one. The other contains fixes exposed by enabling W4 on MSVC which I then rebased this on it too see how many warnings went away. I'll handle the remaining ones later.

**Pull request merged**